ingestion/src/metadata/ingestion/source/database/databricks/unity_catalog/metadata.py

```diff
@@ -0,0 +1,310 @@
+            except Exception as exc:
+                error = f"Unexpected exception to get table [{table.Name}]: {exc}"
+                logger.debug(traceback.format_exc())
+                logger.warning(error)
+                self.status.failed(table.Name, error, traceback.format_exc())
+
+    def yield_table(
+        self, table_name_and_type: Tuple[str, str]
+    ) -> Iterable[Optional[CreateTableRequest]]:
+        """
+        From topology.
+        Prepare a table request and pass it to the sink
+        """
+        table_name, table_type = table_name_and_type
+        table = self.context.table_data
+        schema_name = self.context.database_schema.name.__root__
+        db_name = self.context.database.name.__root__
+        table_constraints = None
+        try:
+            columns = self.get_columns(table.columns)
+
+            table_request = CreateTableRequest(
+                name=table_name,
+                tableType=table_type,
+                description=table.comment,
+                columns=columns,
+                tableConstraints=table_constraints,
+                databaseSchema=self.context.database_schema.fullyQualifiedName,
+            )
+            yield table_request
+
+            if table_type == TableType.View or table.view_definition:
+                self.context.table_views.append(
+                    TableView(
+                        table_name=table_name,
+                        schema_name=schema_name,
+                        db_name=db_name,
+                        view_definition=(
+                            f'CREATE VIEW "{db_name}"."{schema_name}"'
+                            f'."{table_name}" AS {table.view_definition}'
+                        ),
+                    )
+                )
+
+            self.register_record(table_request=table_request)
+        except Exception as exc:
+            error = f"Unexpected exception to yield table [{table_name}]: {exc}"
+            logger.debug(traceback.format_exc())
+            logger.warning(error)
+            self.status.failed(table_name, error, traceback.format_exc())
+
+    def prepare(self):
+        pass
+
+    def get_columns(self, column_data: List[ColumnInfo]) -> Optional[Iterable[Column]]:
+        # process table regular columns info
+
+        for column in column_data:
+            if column.type_text.lower().startswith("union"):
+                column.type_text = column.Type.replace(" ", "")
+            parsed_string = ColumnTypeParser._parse_datatype_string(  # pylint: disable=protected-access
+                column.type_text.lower()
+            )
+            parsed_string["name"] = column.name[:64]
+            parsed_string["dataLength"] = parsed_string.get("dataLength", 1)
+            parsed_string["description"] = column.comment
+            yield Column(**parsed_string)
+
+    def standardize_table_name(self, _: str, table: str) -> str:
+        return table[:128]
```

Comment:
this should not be needed after the FQN hashing. Not sure if it makes sense to keep it here or if we can just remove it and forget

Reply:
cool 👍